OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -82,13 +82,30 @@ public enum ImpactType
 
 	public class AttackInfo
 	{
-		public int Damage;
+		public Damage Damage;
 		public Actor Attacker;
-		public IWarhead Warhead;
 		public DamageState DamageState;
 		public DamageState PreviousDamageState;
 	}
 
+	public class Damage
+	{
+		public int Value;
```

Comment:
https://github.com/OpenRA/OpenRA/pull/11513/files#diff-e107107188001ae1afb408ff1ad5821cR148

I can make the types readonly though.